src/libsync/creds/keychainchunk.h

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (C) by Michael Schuster <michael@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#pragma once
+#ifndef KEYCHAINCHUNK_H
+#define KEYCHAINCHUNK_H
+
+#include <QObject>
+#include <keychain.h>
+#include "accountfwd.h"
+
+// We don't support insecure fallback
+// #define KEYCHAINCHUNK_ENABLE_INSECURE_FALLBACK
+
+namespace OCC {
+
+namespace KeychainChunk {
+
+/*
+* Workaround for Windows:
+*
+* Split the keychain entry's data into chunks of 2048 bytes,
+* to allow 4k (4096 bit) keys / large certs to be saved (see limits in webflowcredentials.h)
+*/
+static constexpr int ChunkSize = 2048;
+static constexpr int MaxChunks = 10;
+
+/*
+ * @brief: Abstract base class for KeychainChunk jobs.
+ */
+class OWNCLOUDSYNC_EXPORT Job : public QObject
+{
+    Q_OBJECT
+public:
+    Job(QObject *parent = nullptr);
+
+    virtual ~Job();
+
+    const QKeychain::Error error() const;
+    const QString errorString() const;
+
+    QByteArray binaryData() const;
+    QString textData() const;
+
+    const bool insecureFallback() const;
```

Comment:
Dunno if I missed it in previous iterations or it appeared only now but s/const bool/bool/

Reply:
Oh strange, I thought I'd have changed that, apparently not in the current push ^^

Reply:
Done

Reply:
Note that I forgot to point it out but that'd apply to `error()` and `errorString()` a couple of lines above.

Reply:
Damn - fixed ;)